ambry-protocol/src/main/java/com.github.ambry.protocol/ReplicaMetadataResponseInfo.java

```diff
@@ -124,14 +127,27 @@ public long sizeInBytes() {
         + +partitionId.getBytes().length + Error_Size_InBytes;
   }
 
+  /**
+   * @return the cumulative size of all the messages represented by this response. 0 if the response signifies an error
+   */
+  public long getTotalSizeOfAllMessages() {
+    return totalSizeOfAllMessages;
+  }
+
   @Override
   public String toString() {
     StringBuilder sb = new StringBuilder();
     sb.append(partitionId);
     sb.append(" ServerErrorCode=").append(errorCode);
     if (errorCode == ServerErrorCode.No_Error) {
+      int size = messageInfoAndMetadataListSerde.getMessageInfoList().size();
```

Comment:
I would extract messageInfoAndMetadataListSerde.getMessageInfoList() into a local variable since it is referenced a few times.

Reply:
Done